include/Piece.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include <iostream>
+#include <string>
+#include "Color.h"
+#include "PieceName.h"
+
+class Board;
+
+// Piece abstract class
+class Piece {
+
+protected:
+	// private variables
+	PieceName _name;
```

Comment:
this is useless you can use another ways to achieve that, like getting the name of the class, you don't have to hold it